tests/e2e/test_save.py

```diff
@@ -0,0 +1,43 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import torch
+
+from azstoragetorch.io import BlobIO
+
+
+@pytest.fixture(scope="module", autouse=True)
+def torch_hub_cache(tmp_path_factory):
+    current_dir = torch.hub.get_dir()
+    torch.hub.set_dir(tmp_path_factory.mktemp("torch_hub"))
+    yield
+    torch.hub.set_dir(current_dir)
+
+
+@pytest.fixture(scope="module")
+def model():
+    model = torch.hub.load("pytorch/vision:v0.10.0", "resnet101", pretrained=False)
+    return model
+
+
+@pytest.fixture(scope="module")
+def state_dict_blob_url(account_url, container_client, tmp_path_factory):
+    model_name = tmp_path_factory.mktemp("model") / "model.pth"
```

Comment:
It might make sense to make the blob name randomized here since we already use this name for the torch load versions of the tests. Mainly by randomizing the name here, we guarantee that we are writing to the blob from a clean slate and we do not run into situations where even if the write failed, the tests pass because a previous fixture had uploaded the model already.